bridge/src/lib.rs

```diff
@@ -91,17 +92,21 @@ methods!(
         let request = unwrap_bytes(request.map_err(VM::raise_ex).unwrap());
         let metadata = to_hash_map(metadata.map_err(VM::raise_ex).unwrap());
         let timeout = timeout.map_or(None, |v| Some(v.to_u64()));
+        let (tx, rx) = mpsc::channel::<RpcResult>();
 
-        let result = Thread::call_without_gvl(move || {
+        let result = Thread::call_without_gvl(|| {
             let connection = _rtself.get_data_mut(&*CONNECTION_WRAPPER);
             let params = RpcParams {
                 rpc: rpc.clone(),
                 request: request.clone(),
                 metadata: metadata.clone(),
                 timeout_millis: timeout
             };
-            connection.call(params)
-        }, Some(|| {}));
+            connection.call(params, tx.clone());
+            rx.recv().expect("RPC result channel closed")
+        }, Some(|| {
+            tx.send(Err(ConnectionError::RequestCancelled)).expect("Unable to close connection");
```

Comment:
You can probably just `tokio::spawn` before the this thread start, and wait on it in the first lambda and https://docs.rs/tokio/latest/tokio/task/struct.JoinHandle.html#method.abort in the second lambda.

Or you can use https://docs.rs/tokio-util/latest/tokio_util/sync/struct.CancellationToken.html. This is what I use in .NET.

Reply:
Nice 👍 Implemented the change using `runtime.block_on` and a cancellation token.

I can't use `tokio::spawn` since I'm calling this from a non-async function and can't `.await`. But block_on works pretty much the same.